python/grass/gunittest/loader.py

```diff
@@ -117,6 +118,7 @@ def discover_modules(
             full = os.path.join(root, testsuite_dir)
 
             all_files = os.listdir(full)
+            files = [f for f in all_files]
```

Comment:
```suggestion
            files = all_files.copy()
```

Is probably better. Copy is all we need.

https://docs.python.org/3/library/stdtypes.html#mutable-sequence-types

Reply:
Applied all three filters